### PR TITLE
Update AspectDefinition to use ToolchainTypeRequirement.

### PR DESCRIPTION
**Pull request by katre**
Part of Optional Toolchains (#14726).

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/android/DexArchiveAspect.java [hunk 1]
```diff
@@ -182,8 +183,12 @@ public AspectDefinition getDefinition(AspectParameters params) {
                     // For proto_lang_toolchain rules, where we just want to get at their runtime
                     // deps.
                     ImmutableSet.of(ProtoLangToolchainProvider.class)))
-            .addRequiredToolchains(
-                Label.parseAbsoluteUnchecked(toolsRepository + sdkToolchainLabel))
+            .addToolchainType(
+                ToolchainTypeRequirement.builder()
+                    .toolchainType(
+                        Label.parseAbsoluteUnchecked(toolsRepository + sdkToolchainLabel))
+                    .mandatory(true)
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/rules/android/DexArchiveAspect.java:190]: I'm a little unhappy with the verbosity here. Should I add a shorter version?

One thought:
```
.addToolchainType(ToolchainTypeRequirement.builder(typeLabel))
```

Thoughts?

Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/rules/android/DexArchiveAspect.java:190]: Sure. As currently written I had some trouble identifying when the builder resolves and how it generally fits with the pattern.

Comment (katre) [reply src/main/java/com/google/devtools/build/lib/rules/android/DexArchiveAspect.java:190]: Okay, added them as #14825, once that's in I'll rebase this.

Comment (katre) [reply src/main/java/com/google/devtools/build/lib/rules/android/DexArchiveAspect.java]: Rebased.

**Review by gregestren — APPROVED** [review]

**Pull request closed without merging**